### DISCUSSION TITLE
sqlite database

### DISCUSSION THREAD

**Discussion by toro-** [requarks/wiki#6122] [Help / Questions] [answered]
I am currently running wikijs on docker.

I would like to update the docker container. Due to my own silliness, I did not map any local volumes and then got right in to writing content. 

What do I need to copy out of the running container to make an update work with my existing content?

I thought it would just be the sqlite db file found in the wiki directory (`/wiki/wikidb.sqlite`), but it does not seem to work. I've also tried copying the entire contents of the `/wiki/` directory but this does not seem to work either.

(Once I had copied these files, I was then mapping local storage to the updated container)

**Comment by NGPixel** [comment, marked as answer]
the db.sqlite file is all you need.

**Reply by toro-** [reply]
this is what I thought... I will review and try again!

**Reply by toro-** [reply]
Got it now that the updated container sees the old db (mis-configuration on my part; forgot to specify DB_FILEPATH 🥴🤦‍♂️)

Now getting error stating that there is a migration lock. Have done the suggested unlock command via knex, it reports it was successful, but database still reports as locked when the container runs again. File permissions seem correct, tried running with `-u="root"` but no luck either.

Wondered if there was something I needed to do in the old container to prepare for moving to another container, but don't see anything there either.

Any ideas?

**Reply by toro-** [reply]
I deleted the migration_lock table and recreating it. This in turn lead me on to error indicating that the DB was in readonly mode. Suspect permissions issue.

Did a horrible thing and `chmod 777 wikidb.sqlite` and it works. So I have a definitely have permissions issue.

Checked what UID and GID the user "node" is running under in the old working container (1000 and 1000). Stopped the new containter, `chown` the file to user on the host with same UID and GID, `chmod 600 wikidb.sqlite` and restarted new container. All good now.
